test/rockpaperscissors.js

```diff
@@ -0,0 +1,868 @@
+
+        it("Call reverts when player one tries to collect player two's forfeit", async () => {
+
+            const gameKey = await rcp.createPlayerOneMoveHash(playerOne, playerOneSecretBytes32, PAPER);
+            await rcp.movePlayerOne(gameKey, "10", {from: playerOne, value: 10});
+            const playerTwoMoveHash = await rcp.createPlayerTwoMoveHash(playerTwo, gameKey, playerTwoSecretBytes32, ROCK);
+            await rcp.movePlayerTwo(gameKey, playerTwoMoveHash, {from: playerTwo, value: 10});
+
+            await timeMachine.advanceTimeAndBlock(DAY_24_HOUR_IN_SECS);
+
+            await truffleAssert.reverts(
+                rcp.playerTwoCollectsForfeit(gameKey, {from: playerOne}),
+                INVALID_PLAYER_MSG
+            );
+
+        }).timeout(TIMEOUT);
+
+        it("Call reverts when player one collects forfeit before the game has expired", async () => {
+
+            const gameKey = await rcp.createPlayerOneMoveHash(playerOne, playerOneSecretBytes32, SCISSORS);
+            await rcp.movePlayerOne(gameKey, "10", {from: playerOne, value: 10});
+            const playerTwoMoveHash = await rcp.createPlayerTwoMoveHash(playerTwo, gameKey, playerTwoSecretBytes32, PAPER);
+            await rcp.movePlayerTwo(gameKey, playerTwoMoveHash, {from: playerTwo, value: 10});
+
+            await truffleAssert.reverts(
+                rcp.playerTwoCollectsForfeit(gameKey, {from: playerTwo}),
+                GAME_NOT_EXPIRED_MSG
+            );
+        }).timeout(TIMEOUT);
+    });
+
+    describe('Player hash creation validation', async function() {
+
+        it("Call reverts when player one creates a move hash with a zero bytes secret", async () => {
+
+            await truffleAssert.reverts(
+                rcp.createPlayerOneMoveHash(playerOne, ZERO_BYTES_32, ROCK),
+                SECRET_EMPTY_MSG
+            );
+
+        }).timeout(TIMEOUT);
+
+        it("Call reverts when player one creates a move hash with an invalid move", async () => {
+
+            await truffleAssert.reverts(
+                rcp.createPlayerOneMoveHash(playerOne, playerOneSecretBytes32, 0),
+                INVALID_MOVE_MSG
+            );
+        }).timeout(TIMEOUT);
+
+        it("Call reverts when player two creates a move hash with a zero bytes game key", async () => {
+
+            await truffleAssert.reverts(
+                rcp.createPlayerTwoMoveHash(playerOne, ZERO_BYTES_32, playerOneSecretBytes32, ROCK),
+                "Game key cannot be empty"
+            );
+
+        }).timeout(TIMEOUT);
+
+        it("Call reverts when player two creates a move hash with a zero bytes secret", async () => {
+
+            const gameKey = await rcp.createPlayerOneMoveHash(playerOne, playerOneSecretBytes32, SCISSORS);
+
+            await truffleAssert.reverts(
+                rcp.createPlayerTwoMoveHash(playerTwo, gameKey, ZERO_BYTES_32, ROCK),
+                SECRET_EMPTY_MSG
+            );
+
+        }).timeout(TIMEOUT);
```

Comment:
I am surprised how you need to extend the time out. That does not sound right.

Reply:
I have set it to this and this seems to work 99% of the time:

    const TIMEOUT = 14000000
    
In combination with making time machine use more granular.